### PR TITLE
CONSOLE-4227: refactor deprecated multiselectdropdown

### PR DESCRIPTION
**Pull request by kmcfaul**
Refactors multiselectdropdown to use Patternfly's `MultiTypeaheadSelect` template, which uses our new `Select` components, and brings the template package in. The main benefit of using this package is a lot of the filtering + keyboard logic and setup is handled internally by the template, saving about 250 lines of code. If there is any problem with adding the dependency, lmk and I can refactor without using the template.

This also updates PF to 5.4 minus charts and topology, which will require additional changes to update.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -153,12 +153,13 @@
     "@patternfly/react-charts": "7.2.2",
     "@patternfly/react-component-groups": "^5.2.0",
     "@patternfly/react-console": "5.1.0",
-    "@patternfly/react-core": "^5.3.0",
-    "@patternfly/react-icons": "^5.3.0",
+    "@patternfly/react-core": "5.4.0",
```
Comment (rhamilto) [inline frontend/package.json:156]: Removing the caret [has been problematic in the past](https://github.com/openshift/console/pull/13588#discussion_r1483491836), but does not appear to be here.  @vojtechszocs, WDYT?

Comment (vojtechszocs) [reply frontend/package.json:156]: I'm OK with it as long as the Yarn lock file does not end up with multiple resolutions for these dependencies.

Comment (kmcfaul) [reply frontend/package.json:156]: Adding the caret in the past may have resolved the issue if a dependency needed a newer version of the pinned package (and the rest of the dependencies were also okay with that newer version, so it only pulled in the newer version).

I'm not sure if that is causing the test failure on CI though, having some trouble parsing what is going wrong there.

Comment (rhamilto) [reply frontend/package.json:156]: > I'm not sure what is causing the test failure on CI though, having some trouble parsing what is going wrong there.

If you view [the build log](https://storage.googleapis.com/test-platform-results/pr-logs/pull/openshift_console/14428/pull-ci-openshift-console-master-frontend/1849126074366038016/build-log.txt), you'll see the following:

```
i18n files are not up to date. Run 'yarn i18n' then commit changes.
```

Your changes result in a diff to the i18n files, so they need to be regenerated and the diff committed.

**Comment by logonoff** [comment]
adding labels because this is debt

/label px-approved
/label docs-approved

**Review by logonoff — COMMENTED** [review]
Seems to work okay using a mouse and keyboard, but when creating a Local Volume Set using a keyboard only, interacting with the menu using the <kbd>Enter</kbd> causes the form to submit

**Review by logonoff — APPROVED** [review]
/lgtm

**Comment by yanpzhan** [comment]
Checked on cluster launched against the pr, regression test about dropdown list on many pages passed.
/label qe-approved

**Pull request merged**